packages/uniforms/__suites__/render.tsx

```diff
@@ -37,11 +39,19 @@ export function render(
     formRef: {} as BaseForm<unknown>,
   };
 
-  return renderOnScreen(element, {
+  const originalRender = renderOnScreen(element, {
     wrapper({ children }) {
       return (
         <context.Provider value={contextValue}>{children}</context.Provider>
       );
     },
   });
+
+  const { rerender } = originalRender;
+
+  const rerenderWithProps = (props: Record<any, any>) => {
+    rerender(cloneElement(element, props));
+  };
```

Comment:
[opinion]
This `props: Record<any, any>` worries me. It can lead to overriding props with a different type than the component accepts. However, I know we receive `ReactElement` in the `render` function, so there is no way of inferring the `ComponentProps<T>` type. I don't know what to do here.

The beta React docs also [discourage the use of `cloneElement`](https://beta.reactjs.org/apis/react/cloneElement#alternatives).

RTL, on the other hand, also approaches the problem differently [in their docs](https://testing-library.com/docs/example-update-props/) (and [on this blog](https://www.designcise.com/web/tutorial/how-to-update-the-props-of-a-rendered-component-in-react-testing-library)). However they show an example with rendering just a single component, but we usually render `<AutoForm>{a lot of children}</AutoForm>` which might lead to a lot of repetition.

Reply:
1. You're right - `cloneElement` is discouraged. But not "in general", but when a render prop would be better. Here we're in a different context and I think it's totally fine.
2. I agree that `Record<any, any>` is far too generic. `Record<string, unknown>` would be already better. But `ReactElement` has two type parameters and actually `Props` is the first one, so we should use that.

Reply:
> `ReactElement` has two type parameters and actually `Props` is the first one
Nice!